text/0182-otlp-remote-parent.md

```diff
@@ -0,0 +1,80 @@
+# Export SpanContext.IsRemote in OTLP
+
+Update OTLP to indicate whether a span's parent is remote.
+
+## Motivation
+
+It is sometimes useful to post-process or visualise only entry-point spans: spans which either have no parent (trace roots), or which have a remote parent.
+For example, the Elastic APM solution highlights entry-point spans (Elastic APM refers to these as "transactions") and surfaces these as top-level operations
+in its user interface.
+
+Currently, the only way entry-point spans can be identified is using (lack of) parent ID, and span kind. Relying on span kind can lead to invalid assumptions,
```

Comment:
If you're looking at a complete trace, you could identify "transactions" by grouping spans according to the [`service.name`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md#service) attribute (which is required). If a parent has a different `service.name`, it is very likely a remote parent.

I guess you want to identify spans with remote parents just by looking at isolated spans? If so, it would be good to add this as an explicit requirement (and maybe add the `service.name` based solution as alternative).

Reply:
Thanks, good call. That assumes that there won't be two services with the same name, but that's probably a fair assumption. I believe this is addressed by this suggestion (please let me know otherwise): https://github.com/open-telemetry/oteps/pull/182/files/7a14ec0e23ea13d90cc8f19d14e8f8dd04a68d82#r812340831